testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replication_eventing.py

```diff
@@ -0,0 +1,511 @@
+    10,
+    100,
+])
+def test_replication_access_revoke_event(params_from_base_test_setup, num_of_docs):
+    """
+    @summary:
+    1. Creating Docs in CBL
+    2. Starting one-shot Push Replication and waiting for it finish
+    3. Changing channel of some docs, so that we get access revoked event
+    4. Replicating access revoked and capturing events through listener
+    5. Verifying the access revoke in event captures
+    """
+    sg_db = "db"
+    sg_url = params_from_base_test_setup["sg_url"]
+    sg_admin_url = params_from_base_test_setup["sg_admin_url"]
+    cluster_config = params_from_base_test_setup["cluster_config"]
+    sg_blip_url = params_from_base_test_setup["target_url"]
+    base_url = params_from_base_test_setup["base_url"]
+    sg_config = params_from_base_test_setup["sg_config"]
+    db = params_from_base_test_setup["db"]
+    cbl_db = params_from_base_test_setup["source_db"]
+    sync_gateway_version = params_from_base_test_setup["sync_gateway_version"]
+
+    if sync_gateway_version < "2.5.0":
+        pytest.skip('This test cannnot run with sg version below 2.5')
+    username = "autotest"
+    password = "password"
+    channels = ["ABC"]
+
+    # Create CBL database
+    sg_client = MobileRestClient()
+
+    # Reset cluster to ensure no data in system
+    c = cluster.Cluster(config=cluster_config)
+    c.reset(sg_config_path=sg_config)
+
+    # 1. Creating Docs in CBL
+    db.create_bulk_docs(num_of_docs, "cbl_docs", db=cbl_db, channels=channels)
+
+    sg_client.create_user(url=sg_admin_url, db=sg_db, name=username, password=password, channels=channels)
+    cookie, session = sg_client.create_session(url=sg_admin_url, db=sg_db, name=username)
+    auth_session = cookie, session
+    sync_cookie = "{}={}".format(cookie, session)
+    session_header = {"Cookie": sync_cookie}
+
+    # Replicating docs to SG so that later we can create conflict
+    # Adding Listener for replicator
+    replicator = Replication(base_url)
+    repl_config = replicator.configure(source_db=cbl_db,
+                                       target_url=sg_blip_url,
+                                       continuous=False,
+                                       headers=session_header,
+                                       replication_type="push")
+    repl = replicator.create(repl_config)
+    repl_change_listener = replicator.addReplicatorEventChangeListener(repl)
+
+    # 2. Starting Replication and waiting for it finish
+    replicator.start(repl)
+    replicator.wait_until_replicator_idle(repl)
+    replicator.removeReplicatorEventListener(repl, repl_change_listener)
+    replicator.stop(repl)
+
+    # 3 changing channel of some docs, so that we get access revoked event
+    sg_docs = sg_client.get_all_docs(url=sg_url, db=sg_db, auth=auth_session)["rows"]
+    docs_to_modify = random.sample(sg_docs, (num_of_docs / 10) * 2)
+    for sg_doc in docs_to_modify:
+        sg_client.update_doc(url=sg_url, db=sg_db, doc_id=sg_doc["id"],
```

Comment:
to change channel, did you try using some other channel, rather than having blank channel?

Reply:
Yes, having blank is same as changing other none.